File: src/main/java/ru/serdtsev/homemoney/account/ReserveRepository.java
package ru.serdtsev.homemoney.account;

import org.springframework.data.repository.CrudRepository;
import ru.serdtsev.homemoney.account.model.Reserve;
import ru.serdtsev.homemoney.balancesheet.BalanceSheet;

import java.util.UUID;

public interface ReserveRepository  extends CrudRepository<Reserve, UUID> {
  Iterable<Reserve> findByBalanceSheet(BalanceSheet balanceSheet);
}

File: src/main/kotlin/ru/serdtsev/homemoney/domain/repository/ReserveRepository.kt
package ru.serdtsev.homemoney.domain.repository

import ru.serdtsev.homemoney.domain.model.account.Reserve
import ru.serdtsev.homemoney.domain.model.balancesheet.BalanceSheet
import java.util.*

interface ReserveRepository {
    fun save(domainAggregate: Reserve)
    fun delete(reserve: Reserve)
    fun exists(id: UUID): Boolean
    fun findById(id: UUID): Reserve
    fun findByIdOrNull(id: UUID): Reserve?
    fun findByBalanceSheet(balanceSheet: BalanceSheet): List<Reserve>
}